core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -884,7 +884,7 @@ class HttpGoogleServicesDAO(
   override def getGoogleProject(googleProject: GoogleProjectId): Future[Project] = {
     implicit val service = GoogleInstrumentedService.Billing
     val cloudResManager = getCloudResourceManagerWithBillingServiceAccountCredential
-    retryExponentially(when500orNon404GoogleError)(() =>
+    retryExponentially(throwable => when500orNonExcludedGoogleError(throwable, Set(StatusCodes.NotFound, StatusCodes.Forbidden)))(() =>
```

Comment:
Background:

Terra billing projects used to be created with Google projects that had a matching name in GCP. But this namespace is global, so if someone else in the world had create a Google project called "hello-world", and then someone in Terra tried to create a project called "hello-world", it would fail. Users want to be able to delete these projects, because they've collected a bunch of them over the years and they serve no value. The existing code would attempt to get the Google project so it could delete it. But in this case, Rawls doesn't actually have access to that Google project -- because Rawls didn't create it. It would retry exponentially until the call timed out. This change quickly surfaces the 403 instead of retrying, and then other code has been changed to proceed with the rest of the billing project deletion if there's a 403.